packages/rn-tester/js/examples/ActivityIndicator/ActivityIndicatorExample.js

```diff
@@ -10,46 +10,37 @@
 
 import type {Node} from 'React';
 import {ActivityIndicator, StyleSheet, View} from 'react-native';
-import React, {Component} from 'react';
+import React, {useState} from 'react';
+import {useEffect} from 'react';
 
-type State = {|animating: boolean|};
-type Props = $ReadOnly<{||}>;
-type Timer = TimeoutID;
+function ToggleAnimatingActivityIndicator() {
+  let _timer;
```

Comment:
I don't know that it is safe to keep the timer handle outside of the effectful code. Can this be kept local to the `useEffect()`?

Reply:
I think that not because at least in my understanding the use effect does not maintain states between executions. I changed the timer to be a ref  and this help me to fix the problem of `// eslint-disable-next-line react-hooks/exhaustive-deps`
Let me know what do you think